core/src/http.rs

```diff
@@ -21,15 +22,20 @@ impl HttpErrorResponse {
     /// the response's status code. (see [RFC 7807](https://www.rfc-editor.org/rfc/rfc7807.html))
     pub async fn from_response(response: Response) -> Self {
         let status = response.status();
+
         if let Some(content_type) = response.headers().get(CONTENT_TYPE) {
-            if content_type == PROBLEM_JSON_MEDIA_TYPE {
-                match response.json::<HttpApiProblem>().await {
-                    Ok(mut problem) => {
-                        problem.status = Some(status);
-                        // Unwrap safety: the conversion always succeeds if the status is populated.
-                        return problem.try_into().unwrap();
+            if let Ok(content_type_str) = content_type.to_str() {
```

Comment:
I want to make an observation about using things like `Result::map_err` or `Option::ok_or_else` to move some of this code left, but to be honest it's not immediately obvious to me how to do that since this method doesn't return `Result`. This function was a bit of a mess before you got here, so I think we can live with it.

Reply:
Also resolved in 326436bf..2af7c099.